src/engine.cpp

```diff
@@ -21,35 +23,54 @@ Engine::~Engine() {
 void Engine::start() {
     initialize();
 
-    main_loop();
+    try {
+        main_loop();
+    } catch (std::exception &e) {
+        spdlog::error("Crashed with exception: {}", e.what());
+    } catch (...) {
+        spdlog::error("Dupa");
```

Comment:
The best error message. As it should never be thrown, I agree it should be left.